src/main/java/com/redhat/rhjmc/containerjfr/commands/internal/SaveRecordingCommand.java

```diff
@@ -88,16 +92,29 @@ public boolean isAvailable() {
     private String saveRecording(IRecordingDescriptor descriptor)
             throws IOException, FlightRecorderException, JMXConnectionException {
         String recordingName = descriptor.getName();
-        if (!recordingName.endsWith(".jfr")) {
-            recordingName += ".jfr";
+        if (recordingName.endsWith(".jfr")) {
+            recordingName = recordingName.substring(0, recordingName.length() - 4);
         }
         String targetName = getConnection().getHost().replaceAll("[\\._]+", "-");
-        String destination = String.format("%s_%s", targetName, recordingName);
-        fs.copy(
-            getService().openStream(descriptor, false),
-            recordingsPath.resolve(destination),
-            StandardCopyOption.REPLACE_EXISTING
-        );
+        String timestamp = clock.now().truncatedTo(ChronoUnit.SECONDS).toString().replaceAll("[-:]+", "");
+        String destination = String.format("%s_%s_%s", targetName, recordingName, timestamp);
+        // TODO byte-sized rename limit is arbitrary. Probably plenty since recordings are also differentiated by second-resolution timestamp
+        byte count = 1;
+        while (fs.exists(recordingsPath.resolve(destination + ".jfr"))) {
+            destination = String.format("%s_%s.%d", targetName, recordingName, count++);
+            if (count == Byte.MAX_VALUE) {
+                throw new IOException("Recording could not be saved. File already exists and rename attempts were exhausted.");
+            }
+        }
+        if (!destination.endsWith(".jfr")) {
```

Comment:
This check seems redundant? Looking at L103 and L100, it's impossible for `destination` to end with `.jfr`.

Reply:
Very true. Added and then forgot to remove that in an earlier version of this changeset. I'll pull it out in a minute.